crates/store/src/db/sql/mod.rs

```diff
@@ -178,6 +178,32 @@ pub fn select_account(transaction: &Transaction, account_id: AccountId) -> Resul
     account_info_from_row(row)
 }
 
+/// Select the latest account details by account id from the DB using the given [Connection].
+///
+/// # Returns
+///
+/// The latest account details, or an error.
+pub fn select_account_by_prefix(transaction: &Transaction, id_prefix: u32) -> Result<AccountInfo> {
+    let mut stmt = transaction.prepare_cached(
+        "
+        SELECT
+            account_id,
+            account_commitment,
+            block_num,
+            details
+        FROM
+            accounts
+        WHERE
+            id_prefix = ?1;
```

Comment:
As far as I can tell `id_prefix` field does not exist in the `accounts` table. Is there a commit missing from this PR?

Reply:
Yes, I failed to add it here. Just added those changes

Reply:
`id_prefix` isn't indexed in the DB, this might end up being a really expensive query